NineChronicles.Headless.Tests/GraphTypes/GraphQLTestBase.cs

```diff
@@ -71,46 +63,43 @@ public GraphQLTestBase(ITestOutputHelper output)
                     ),
                 }, blockAction: blockAction);
 
-            var blockPolicy = new BlockPolicy<PolymorphicAction<ActionBase>>(blockAction: blockAction);
-            var blockChain = new BlockChain<PolymorphicAction<ActionBase>>(
-                blockPolicy,
-                new VolatileStagePolicy<PolymorphicAction<ActionBase>>(),
-                store,
-                stateStore,
-                genesisBlock,
-                renderers: new IRenderer<PolymorphicAction<ActionBase>>[] { new BlockRenderer(), new ActionRenderer() }
-            );
-
+            var ncService = ServiceBuilder.CreateNineChroniclesNodeService(genesisBlock, new PrivateKey());
             var tempKeyStorePath = Path.Join(Path.GetTempPath(), Path.GetRandomFileName());
             var keyStore = new Web3KeyStore(tempKeyStorePath);
 
             StandaloneContextFx = new StandaloneContext
             {
-                BlockChain = blockChain,
                 KeyStore = keyStore,
             };
+            ncService.ConfigureContext(StandaloneContextFx);
 
             var configurationBuilder = new ConfigurationBuilder();
             var configuration = configurationBuilder.Build();
 
             var services = new ServiceCollection();
             services.AddSingleton(StandaloneContextFx);
             services.AddSingleton<IConfiguration>(configuration);
+            services.AddSingleton(_ => ServiceBuilder.CreateNineChroniclesNodeService(genesisBlock, new PrivateKey()));
             services.AddGraphTypes();
             services.AddLibplanetExplorer<NCAction>();
             services.AddSingleton<StateQuery>();
+            services.AddSingleton(_ => ncService);
```

Comment:
How about using `ServiceCollection.AddSingleton<TService>(IServiceCollection, TService)`? Because it doesn't use anything from the `_`, `provider` parameter.

Reply:
`.AddSingleton<T>()` is generally proper, but we can't use that because `NineChroniclesNodeService` couldn't be constructed by DI. (only `NineChroniclesNodeService.Create()` allowed yet) we should need to more refactor for it. 😓

Reply:
Oh, yes. As you said, `NineChroniclesNodeService` couldn't be constructed by DI. But what I wanted to suggest was

```suggestion
            services.AddSingleton(ncService);
```

not

```csharp
services.AddSingleton<NineChroniclesNodeService>();
```

How do you think about this?

Reply:
I see, fixed. https://github.com/planetarium/NineChronicles.Headless/pull/481/commits/7f11f097f211230490a54cef8427653fefd61e60